docs/architecture/adr-009-evidence-module.md

```diff
@@ -0,0 +1,222 @@
+
+  return sdk.Result{
+    // ...
+  }
+}
+```
+
+The `x/evidence` module's keeper is responsible for matching the `Evidence` against
+the module's router. Upon success the validator is slashed and the infraction is
+persisted. In addition, the validator is jailed is the `Evidence` type is configured
+to do so.
+
+```go
+func (k Keeper) SubmitInfraction(ctx sdk.Context, infraction Infraction) sdk.Error {
+  handler := keeper.router.GetRoute(infraction.Evidence.Route())
+  if err := handler(cacheCtx, infraction.Evidence); err != nil {
+    return ErrInvalidEvidence(keeper.codespace, err.Result().Log)
+  }
+
+  penalty := keeper.GetSlashingPenalty(ctx, infraction.Evidence.Type())
+
+  keeper.stakingKeeper.Slash(
+    ctx,
+    infraction.ConsensusAddress,
+    infraction.InfractionHeight,
+    infraction.Power,
+    penalty.SlashFraction,
+  )
+
+  if penalty.Jailable {
+    keeper.stakingKeeper.Jail(ctx, infraction.ConsensusAddress)
+  }
+
+  keeper.setInfraction(ctx, infraction)
+}
+```
+
+### Genesis
+
+We require the the `x/evidence` module's keeper to keep an internal persistent
+mapping between `Evidence` types and slashing penalties represented as `InfractionPenalty`.
+
+```go
+var slashingPenaltyPrefixKey = []byte{0x01}
+
+type InfractionPenalty struct {
+  EvidenceType    string
+  Penalty         sdk.Dec
+}
+
+func GetSlashingPenaltyKey(evidenceType string) []byte {
+  return append(slashingPenaltyPrefixKey, []byte(evidenceType)...)
+}
+
+func (k Keeper) GetSlashingPenalty(ctx sdk.Context, evidenceType string) sdk.Dec {
+  store := ctx.KVStore(k.storeKey)
+
+  bz := store.Get(GetSlashingPenaltyKey(evidenceType))
+  if len(bz) == 0 {
+    return sdk.ZeroDec()
+  }
+
+  var ip InfractionPenalty
+  k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &ip)
+
+  return ip.Penalty
+}
+```
+
+Finally, we need to represent the genesis state of the `x/evidence` module. The
+module only needs a list of all submitted valid infractions and the infraction
+penalties which we represent through parameters managed by the `x/params` module.
+
+```go
+type PenaltyParam struct {
+  SlashFraction  sdk.Dec
+  Jailable       bool
+}
+
+type PenaltyParams struct {
+  // A series of misbehaviour penalties all of which are of type PenaltyParam
+  // and these can individually be changed through a governance parameter change
+  // proposal.
+}
+
+type GenesisState struct {
+  Params       PenaltyParams
+  Infractions  []Infraction
+}
+```
+
+## Consequences
+
+### Positive
+
+- Allows the state machine to process equivocations submitted on-chain and penalize
+validators based on agreed upon slashing parameters
+- Does not solely rely on Tendermint to submit evidence
+
+### Negative
+
+- No easy way to introduce new evidence types through governance on a live chain
+due to the inability to introduce the new evidence type's corresponding handler
+
+### Neutral
+
+- Should we persist infractions indefinitely? Or should we rather rely on events?
+
+## References
```

Comment:
we should also add ICS03 as a reference here

Reply:
2 or 3? As the link above routes to ics-003

Reply:
ICS02 sorry